server/mongodb/actions/ForgotPassword.js

```diff
@@ -0,0 +1,58 @@
+import bcrypt from "bcryptjs";
+import crypto from "crypto";
+import jwt from "jsonwebtoken";
+import mongoDB from "../index";
+import PasswordResetRequest from "../models/PasswordResetRequest";
+import User from "../models/User";
+import { sendEmail } from "../../actions/email.js";
+import urls from "../../../utils/urls.js";
+
+export async function forgotPassword(email) {
+  if (email == null) {
+    throw new Error("Email must be provided");
+  }
+
+  await mongoDB();
+
+  try {
+    const token = crypto.randomBytes(128).toString();
```

Comment:
This looks like it generates special symbols in the token which might cause issues down the line-- can you switch this to use UUIDs?